### PR TITLE
scitools: init at 0.9.0

### PR DESCRIPTION
**Pull request by jamtrott**
###### Motivation for this change
Add the scitools Python package, which contains many useful tools for scientific computing in Python. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/scitools/default.nix
Comment (vyp) [inline pkgs/development/python-modules/scitools/default.nix]: This can be 'simplified' by using `fetchFromGitHub` instead:

``` nix
{
  src = fetchFromGitHub {
    owner = "hplgit";
    repo = "scitools";
    rev = "${name}";
    sha256 = "...";
  };
}
```

Just run `nix-prefetch-url --unpack 'https://github.com/hplgit/scitools/archive/scitools-0.9.0.tar.gz'` to get the sha256 (if it's different).

File: pkgs/development/python-modules/scitools/default.nix
Comment (vyp) [inline pkgs/development/python-modules/scitools/default.nix]: Add the license under meta here with `license = stdenv.lib.licenses.bsd3;`. And also add yourself as maintainer, `maintainers = with stdenv.lib.maintainers; [ jamtrott ];` after adding yourself to https://github.com/NixOS/nixpkgs/blob/master/lib/maintainers.nix.

**Review by vyp — CHANGES_REQUESTED** [review]

**Comment by jamtrott** [comment]
@vyp, Thanks for the review!  The above two patches should address the issues.

File: pkgs/development/python-modules/scitools/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/scitools/default.nix]: also propagated

**Review by FRidh — CHANGES_REQUESTED** [review]
Judging from the setup.py this package only works with Python 2.x.

**Comment by veprbl** [comment]
@jamtrott Are you still interested in implementing this? Before this is merged you would need to address remaining issues like using `propagatedBuildInputs` for `matplotlib` and setting `disabled = !isPy27` and do a rebase.

**Comment by jamtrott** [comment]
Hi, @veprbl.  I don't need this package anymore, and I'm afraid that it's no longer being maintained anyway.  I will close the pull request.

Thank you all for your help.

**Pull request closed without merging**
